pf_driver/src/pf/pf_packet.cpp

```diff
@@ -14,7 +14,7 @@ bool PFPacket::parse_buf(uint8_t* buf, size_t buf_len, size_t &remainder, size_t
     std::tie(h_size, p_size, num) = read_header(stream);
 
     auto data_size = p_size - h_size;
-    if((buf_len - SIZE) < data_size)
+    if(buf_len < p_size)
```

Comment:
This comment concerns me:
```
so we'll end up with some header fields that might be invalid (with no indication to the user)
```
Also I wanted to move all the parsing related checks in `pf_parser.h`. Because that is what the parser class was supposed to do. This commit might be a good place to do that.

Reply:
WRT the comment about header fields being invalid, any thoughts on how you *would* like to handle that?  I guess it seems to me like there are basically four options:
1. Since those header fields simply aren't supplied by the sensor, their value won't have meaning anyway, and we don't worry too much about what value ends up in them.  (this is the current approach)
2. Those header fields should be set to zero value.  This at least avoids giving the user the impression that the fields are somehow working or weirdly broken (because they have changing data in them), but doesn't necessarily make it obvious to the user that the fields *aren't* working.  This approach would probably be reasonably easy by zeroing portions of the header before passing it to `ros::serialization`.
3. Fill fields with a non-zero value that indicates "no data" (not sure if there is even such a value).  This gets much harder because we then end up duplicating work that's being done now in ros serialization in order to figure out *which* header fields need this special value.
4. There might be some way to alter the header in realtime to indicate that these types aren't even present, but again, that seems like it becomes even more specialized, and might necessitate further conditionals in code that uses the header data, since it would then have to check packet types or header presence or something.

I guess I'd say either option 1 (as is) or option 2 (easy change to zero part of a buffer).

Reply:
WRT moving parsing checks to `pf_parser.h`, I'm not sure how much more makes sense to do there.  I see four checks total:
1. (in parser) check that the buffer has at least enough bytes for a header
2. (in parser) check that the packet starts with the right magic number
3. (in `packet.parse_buf`) check that the buffer has at least enough bytes for the full packet
4. (in `packet.parse_buf`) check that the buffer has at least enough bytes for all the data points expected

Checks 3 and 4 are kinda redundant (and 4 currently assumes 32 bits per point, which isn't right for packet type B), so I'd personally be inclined to drop check 4 altogether.

Check 3 is currently being done here because it requires having parsed the packet header to know how many bytes are *supposed* to be in the packet.  I guess technically, parsing the packet header could be moved to the `pf_parser.h`, but if so, we might as well move this entire function to `pf_parser.h`, since it would only be four lines long at that point.

I guess that's a long way of saying that I'd be inclined to leave this check in `packet.parse_buf` and maybe delete the fourth check about number of points.

Reply:
> I guess that's a long way of saying that I'd be inclined to leave this check in packet.parse_buf and maybe delete the fourth check about number of points.

Agreed.

Reply:
Okay, deleting the fourth check (enough data for expected number of points).

Still haven't taken any action on possibly invalid header fields.  I was thinking I'd just implement option 2 there (and zero portions of the header before passing to `ros::serialization`, but when I went to do it I realized that the correct header length is one of the parameters we get by deserializing the header, so we might end up doing it twice or something.  Not happy with any of the choices there.